### DISCUSSION TITLE
Welcome to nr1-github Discussions!

### DISCUSSION THREAD

**Discussion by shahramk** [newrelic/nr1-github#98] [Announcements]
## 👋 Welcome!
Hello from New Relic Labs - we are the maintainers of the [nr1-github](../) repo, and we're using Discussions as a place to connect directly with members of our community.

If you have questions to ask or ideas you'd like to share, please feel free to start a discussion.

If you'd like to report a bug, or submit a feature request, go ahead and open up an issue.

We are a small team so we might not be able to respond right away, but rest assured we are keeping an eye on this space - your input is important to us!

[<img src="https://user-images.githubusercontent.com/1786630/214122263-7a5795f6-f4e3-4aa0-b3f5-2f27aff16098.png" height=100/>](https://github.com/newrelic?q=nrlabs-viz&type=all&language=&sort=)

**Comment by monjie110** [comment]
docker run \
 -d \
 --name newrelic-infra \
 --network=host \
 --cap-add=SYS_PTRACE \
 --privileged \
 --pid=host \
 -v "/:/host:ro" \
 -v "/var/run/docker.sock:/var/run/docker.sock" \
 -e  NRIA_LICENSE_KEY=e9bd38ce9dbffd4ed6431a7bf0306537bbf9NRAL \
 newrelic/infrastructure:latest

**Comment by monjie110** [comment]
import numpy as np
import pandas as pd
from sklearn.preprocessing import MinMaxScaler
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, LSTM, Dropout, BatchNormalization

# Load CSV data
def load_csv_data(csv_file_path):
    data = pd.read_csv(csv_file_path)
    return data

# Preprocess data
def preprocess_data(data, columns_to_drop):
    # Drop null values and unimportant columns
    data = data.dropna()
    data = data.drop(columns_to_drop, axis=1)

    # Scale the data
    scaler = MinMaxScaler()
    data = scaler.fit_transform(data)

    # Split data into training and testing sets
    train_size = int(len(data) * 0.8)
    train_data = data[:train_size]
    test_data = data[train_size:]

    # Reshape data for LSTM input
    x_train, y_train = [], []
    for i in range(60, len(train_data)):
        x_train.append(train_data[i-60:i])
        y_train.append(train_data[i])
    x_test, y_test = [], []
    for i in range(60, len(test_data)):
        x_test.append(test_data[i-60:i])
        y_test.append(test_data[i])
    x_train, y_train = np.array(x_train), np.array(y_train)
    x_test, y_test = np.array(x_test), np.array(y_test)

    return x_train, y_train, x_test, y_test, scaler

# Build LSTM model
def build_model(input_shape):
    model = Sequential()
    model.add(LSTM(128, return_sequences=True, input_shape=input_shape))
    model.add(BatchNormalization())
    model.add(Dropout(0.2))
    model.add(LSTM(64, return_sequences=True))
    model.add(BatchNormalization())
    model.add(Dropout(0.2))
    model.add(LSTM(32, return_sequences=False))
    model.add(BatchNormalization())
    model.add(Dense(25))
    model.add(Dense(1))
    return model

# Train LSTM model
def train_model(model, x_train, y_train):
    model.compile(optimizer='rmsprop', loss='mean_squared_error')
    model.fit(x_train, y_train, batch_size=16, epochs=50)

# Predict stock price and decide whether to buy or sell
def predict_price(model, x_test, scaler, last_known_price):
    # Make predictions on test data
    predictions = model.predict(x_test)
    predictions = scaler.inverse_transform(predictions)

    # Get the last known closing price
    last_known_price = scaler.inverse_transform(np.array([[last_known_price]]))

    # Determine whether to buy or sell
    if predictions[-1] > last_known_price:
        decision = "buy"
    else:
        decision = "sell"

    return predictions[-1][0], decision

# Evaluate model accuracy with RMSE metric
def evaluate_model(model, x_test, y_test, scaler):
    # Make predictions on test data
    predictions = model.predict(x_test)
    predictions = scaler.inverse_transform(predictions)

    # Calculate RMSE
    rmse = np.sqrt(np.mean(((predictions - y_test) ** 2)))

    return rmse

# Add noise to data for data augmentation
def add_noise(data, noise_factor):
    noisy_data = data + noise_factor * np.random.randn(*data.shape)
    return noisy_data

# Load and preprocess data
data = load_csv_data('stock_prices.csv')
columns_to_drop = ['Date', 'Open', 'High', 'Low', 'Adj Close', 'Volume']
x_train, y_train, x_test, y_test, scaler = preprocess_data(data, columns_to_drop)

# Data augmentation by adding noise
x_train_noisy = add_noise(x_train, 0.1)

# Build and train model
model_input_shape = (x_train.shape[1], x_train.shape[2])
model = build_model(model_input_shape)
train_model(model, x_train_noisy, y_train)

# Evaluate model accuracy
rmse = evaluate_model(model, x_test, y_test, scaler)
print("RMSE:", rmse)

# Predict future stock prices
last_known_price = data['Close'].iloc[-1]
future_prices = []
for i in range(10):
    future_price, decision = predict_price(model, np.array([x_test[-1]]), scaler, last_known_price)
    future_prices.append(future_price)
    if decision == "buy":
        print("Buy")
    else:
        print("Sell")
    last_known_price = future_price

# Print predicted future prices
print("

**Reply by monjie110** [reply]
docker run \
 -d \
 --name newrelic-infra \
 --network=host \
 --cap-add=SYS_PTRACE \
 --privileged \
 --pid=host \
 -v "/:/host:ro" \
 -v "/var/run/docker.sock:/var/run/docker.sock" \
 -e  NRIA_LICENSE_KEY=e9bd38ce9dbffd4ed6431a7bf0306537bbf9NRAL \
 newrelic/infrastructure:latest

**Reply by EQemccudden** [reply]
@monjie110 - I'd advise removing your keys as per [here](https://docs.newrelic.com/docs/apis/intro-apis/new-relic-api-keys/#security-practices)
